data/spells/scripts/monster/souleater_wave.lua

```diff
@@ -0,0 +1,15 @@
+area = {
+	{0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0},
+	{0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0},
+	{0, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0},
+	{0, 0, 0, 0, 0, 3, 0, 0, 0, 0, 0}
+}
+
+local combat = Combat()
+combat:setParameter(COMBAT_PARAM_TYPE, COMBAT_LIFEDRAINDAMAGE)
+combat:setParameter(COMBAT_PARAM_EFFECT, CONST_ME_SOUND_RED)
+combat:setArea(createCombatArea(area))
+
+function onCastSpell(creature, var)
+	return combat:execute(creature, var)
+end
```

Comment:
this spell is not needed, souleater already have this spell coded on his xml file

```
		<attack name="death" interval="2000" chance="10" length="4" spread="3" target="0" min="-100" max="-200">
			<attribute key="areaEffect" value="rednote" />
		</attack>
```